events.go

```diff
@@ -214,6 +216,41 @@ var fakeCancelReason = []byte("internal: cancel")
 var fakeTimeoutReason = []byte("internal: call state timeout")
 var fake2xxReason = []byte("internal: implied OK")
 
+func (d *EventData) EncryptIP(key [16]byte) error {
+	var err error
+	if d.Src.To4() != nil && d.Dst.To4() != nil {
+		err = d.EncryptIPv4(key)
+	} else if d.Src.To16() != nil && d.Dst.To16() != nil {
+		err = d.EncryptIPv6(key)
+	} else {
+		err = errors.New("EncryptIP: broken IP address")
+	}
+	return err
+}
+
+// EncryptIP encrypts source and destination IP addresses using a format preserving encryption algorithm (ipcrypt)
+func (d *EventData) EncryptIPv4(key [16]byte) error {
+	var c [4]byte
+	var err error
+	if c, err = ipcrypt.EncryptBin(key, d.Src); err != nil {
+		return err
+	}
+	for i, _ := range c {
+		d.Src[12+i] = c[i]
+	}
```

Comment:
```suggestion
	copy(d.Src, c)
```
d.Src is a slice pointing inside the big event buf (d.Buf) and is size depends on the ip. For IPv4 is always 4 bytes. There is no place to extend it, so you always have to copy only max len(d.Src), in this case 4.